### PR TITLE
Attach user token to generator added

### PR DESCRIPTION
**Pull request by annyhe**
- include test

### CODE REVIEW

File: tests/api/v1/collectors/heartbeat.js [hunk 1]
```diff
@@ -1017,7 +1018,43 @@ describe('tests/api/v1/collectors/heartbeat.js >', () => {
         .then(done)
         .catch(done);
       });
+    });
+
+    describe('added generator >', () => {
+      before(() => tu.toggleOverride('returnUser', true));
+      after(() => tu.toggleOverride('returnUser', false));
```
Comment (iamigo) [inline tests/api/v1/collectors/heartbeat.js]: "after" should restore it to whatever it was before, which is not necessarily guaranteed to be "false" -- safer to save the original value in the "before"

Comment (annyhe) [reply tests/api/v1/collectors/heartbeat.js]: We are not using the returnUser anywhere else in the test. Even if we are, the other tests ought to setup then teardown. Not sure if it is a good idea to add more code to this section.

Comment (jgraff2) [reply tests/api/v1/collectors/heartbeat.js]: don't think it matters much either way.

File: tests/api/v1/collectors/heartbeat.js [hunk 2]
```diff
@@ -1017,7 +1018,43 @@ describe('tests/api/v1/collectors/heartbeat.js >', () => {
...
+          }
 
+          expect(res.body.createdBy).to.equal(userId);
+          const timestamp = Date.now();
+          return Promise.resolve()
+          .then(() => postGenerator(generator2, [collector1]))
+          .then(() => sendHeartbeat(collector1))
+          .then((res) => {
+            const reencryptedSG = res.body.generatorsAdded[0];
+            expect(reencryptedSG.token).to.be.an('string');
```
Comment (jgraff2) [inline tests/api/v1/collectors/heartbeat.js]: to.be.a('string')

Comment (jgraff2) [inline tests/api/v1/collectors/heartbeat.js]: shouldn't you also check that the value of the token is correct?

Comment (annyhe) [reply tests/api/v1/collectors/heartbeat.js]: good catch. Updated

Comment (annyhe) [reply tests/api/v1/collectors/heartbeat.js]: do you mean check it is a proper UUID, or the expected UUID? The latter is difficult as the jwtUtil token generation is one way.

Comment (jgraff2) [reply tests/api/v1/collectors/heartbeat.js]: ok, makes sense.

Comment (jgraff2) [reply tests/api/v1/collectors/heartbeat.js]: If you updated it, doesn't look like it got pushed.

Comment (annyhe) [reply tests/api/v1/collectors/heartbeat.js]: @jgraff2 how do I check the value of the token is correct? Do I use jwtUtil.verifyToken(req, callback)? or make a new function in jwtUtil with 
  return jwtVerifyAsync(token, secret, {})
  .then((payLoad) => {
    decodedData = payLoad;
    return User.findOne({ where: { name: decodedData.username } });
  })
  .then((user) => {
    if (!user) {
      throw new apiErrors.ForbiddenError({
        explanation: 'Invalid Token.',
      });
    }

    req.user = user.get();

Comment (annyhe) [reply tests/api/v1/collectors/heartbeat.js]: @jgraff2 in retrospect that is a lot of work to check it is a token. Any suggestions?

Comment (jgraff2) [reply tests/api/v1/collectors/heartbeat.js]: I think it's fine as-is, given that the token is being created just before this and createToken() is tested elsewhere.

File: tests/api/interim.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+const expect = require('chai').expect;
```
Comment (jgraff2) [inline tests/api/interim.js]: Looks like you accidentally included code from a different story.

File: tests/api/v1/collectors/heartbeat.js (hunk 1 above)
Comment (iamigo) [reply tests/api/v1/collectors/heartbeat.js]: It's not about whether it's used in *this* test, but rather across all the tests.

**Pull request merged**